src/xdocs/idea.xml

```diff
@@ -43,7 +43,9 @@
 
     <section name="Debug">
       <p>
-        Open the Check's source file by double-click on it in a source tree as is shown:<br/><br/>
+        Open the Check's source file by double-click on it in a source tree as shown
+        in the figure below. Please note that the path to the file shown in the
+        figure is <code>src/test/java/checks</code><br/><br/>
```

Comment:
I don't feel this alleviates confusion.
> note that the path to the file shown in the figure 

Image right below still shows main folder and not test. If you read full documentation, this section is saying to open main file to set a breakpoint. It is later down that it says open test and right click that to actually run and debug the test.

I think we need a new first paragraph. Explain this entire section, in short, you need to open main file to set breakpoint you wish and then open test file to run test. In this new paragraph, I would then explain tests are in test folder and main are in main folder. I would then  restore this original paragraph to it's original.

Reply:
I think what your saying is missing is we need another in the list with a picture that basically says "Open the Test's source file by double-click on it in a source tree as is shown:".

I don't think the first image should change as we are first explaining that you need to set a breakpoint in the required place. If you debug without setting a breakpoint, it will just run all the way through and not stop.

If you agree, then feel free to make the change.

Reply:
@rnveach Let me try to present my point in more systematic way.

Documentation is really good, I agree. I went into some confusion and I believe many newbies who will contribute to checkstyle can go into this confusion as well. Here's how:


After the line `Then right-click the corresponding Unit-test file or class definition > Debug "testName"`, there is an image below which shows right clicking on Test file. The confusion that arises here is that: I thought the file is DescendantTokenCheck.java on which debug option is available/breakpoint since at the top corner (see in image) you can see that DescendantTokenCheck.java is opened and not DescendantToeknCheckTest.java

Reply:
![checkstyleImage](https://user-images.githubusercontent.com/86051118/195655943-2ee587e6-cf21-4b2a-a29e-2ac2de2e368f.png)

Reply:
@romani WDYT?

Reply:
I am fine with change of image

Reply:
@NitishKumar06 I am following the issue.

Your confusion started at `Then right-click the corresponding Unit-test file`. So it doesn't make sense to change things before this line as you said you weren't confused with them and they are the needed pre-cursor to start debugging. There is no point to run in debug mode without a breakpoint.

I still think we need another step in the list with a picture that basically says "Open the Test's source file by double-click on it in a source tree as is shown:". There can be an accompanying picture with it.

Reply:
Yes @rnveach ! This sounds good.

Reply:
@NitishKumar06 I am unsure why you requested a review when my requested changes are not done.

Reply:
Could you please mention which is the right folder to store the pictures as I'm looking forward to modify the picture

Reply:
https://github.com/checkstyle/checkstyle/blob/master/src/xdocs/idea.xml#L48
https://github.com/checkstyle/checkstyle/blob/master/src/site/resources/images/gui_screenshot_select_check_idea.png

Reply:
@rnveach I have done almost the changes you've mentioned. After generating the site locally, I found that the image is too big which won't give a user a good experience. How do I modify this image to have its size the same as the above images?

![errorImage](https://user-images.githubusercontent.com/86051118/198917222-45495f0a-691f-4185-ac39-b6d16f781a72.png)

Reply:
Most likely you will have to shrink it in an image editing software.

Reply:
Got it! I'd be applying the changes then. Any more changes apart from that required from your end?